// bb-media-selector/webpack.config.js
const path = require('path');
const webpack = require('webpack');
module.exports = {
    context: path.resolve(__dirname, './src'),
    entry: {
        app: './index.js',
    },
    output: {
        path: path.join(__dirname, '/lib/'),
        filename: 'index.js',
        library: 'shared-components',
        libraryTarget: 'umd',
        umdNamedDefine: true
    },
    devServer: {
    contentBase: path.resolve(__dirname, './src'),  // New
  },
    module: {
			rules: [
				{
					test: /\.js$/,
					exclude: [/node_modules/],
					use: [
						{
							loader: 'babel-loader',
							options: {
								presets: [
										['es2015', { modules: false }],
										'react',
										'stage-2',
										'stage-3'
									],
								plugins: [
									'syntax-decorators',
								],
							},
						}],
				},
				{
					test: /\.(ttf|eot|woff|woff2|svg|gif|jpg|png)$/,
					use: 'url-loader?limit=10000',
				},
				{
					test: /\.css$/,
					include: /components\/elements/,
					use: [
						'css-module-file-loader',
						// 'css-loader?modules&localIdentName=_[hash:base64:6]',
						// 'css-loader?modules&localIdentName=pb-[local]',
						'css-loader?modules&localIdentName=[local]',
						'postcss-loader',
						'element-css-loader',
					],
				},
				{
					test: /\.scss$/,
					include: /components\/elements/,
					use: [
						'css-module-file-loader',
						'css-loader?modules&minimize&localIdentName=_[hash:base64:10]',
						'postcss-loader',
						'sass-loader',
					],
				},
				{
					test: /\.css/,
					include: /styles\/vendor|node_modules/,
					use: [
						'style-loader',
						'css-loader',
						'postcss-loader',
					],
				},
				{
					test: /\.css$/,
					exclude: /node_modules|vendor|components\/elements/,
					use: [
						'style-loader',
						'css-loader?modules&minimize&localIdentName=_[hash:base64:10]',
						'postcss-loader',
					],
				},
				{
					test: /\.scss$/,
					exclude: /node_modules|vendor|components\/elements/,
					use: [
						'style-loader',
						'css-loader?modules&minimize&localIdentName=_[hash:base64:10]',
						'sass-loader',
					],
				},
			],
		},
};
